frontend/src/metabase/browse/components/ModelGroup.tsx

```diff
@@ -1,34 +1,59 @@
-import Link from "metabase/core/components/Link";
-import Search from "metabase/entities/search";
+import { useMemo } from "react";
+import { t, c, msgid } from "ttag";
+
 import { color } from "metabase/lib/colors";
-import { useDispatch } from "metabase/lib/redux";
 import * as Urls from "metabase/lib/urls";
-import { Box, Group, Icon, Text, Title } from "metabase/ui";
+import { Box, Icon, Title, Button, Flex } from "metabase/ui";
 import type {
   Card,
-  CollectionEssentials,
   SearchResult,
+  CollectionEssentials,
 } from "metabase-types/api";
 
-import { getCollectionName, sortModels } from "../utils";
+import { getCollectionName, sortModels, getIcon } from "../utils";
 
 import {
+  CollectionCollapse,
+  CollectionExpandCollapseContainer,
   CollectionHeaderContainer,
-  CollectionHeaderGroup,
-  CollectionHeaderLink,
+  CollectionHeaderToggleContainer,
+  CollectionSummary,
+  FixedSizeIcon,
   ModelCard,
+  ModelCardLink,
   MultilineEllipsified,
+  HoverUnderlineLink,
 } from "./BrowseModels.styled";
-import { LastEdited } from "./LastEdited";
+
+const MAX_COLLAPSED_MODELS = 6;
 
 export const ModelGroup = ({
   models,
   localeCode,
+  expanded,
+  showAll,
+  toggleExpanded,
+  toggleShowAll,
 }: {
   models: SearchResult[];
   localeCode: string | undefined;
+  expanded: boolean;
+  showAll: boolean;
+  toggleExpanded: () => void;
+  toggleShowAll: () => void;
 }) => {
-  const sortedModels = models.sort((a, b) => sortModels(a, b, localeCode));
+  const { aboveFoldModels, belowFoldModels } = useMemo(() => {
+    const sortedModels = models.sort((a, b) => sortModels(a, b, localeCode));
+
+    return {
+      aboveFoldModels: sortedModels.slice(0, MAX_COLLAPSED_MODELS),
+      belowFoldModels: sortedModels.slice(MAX_COLLAPSED_MODELS),
+    };
+  }, [models, localeCode]);
+
+  const visibleModels = showAll
+    ? [...aboveFoldModels, ...belowFoldModels]
+    : aboveFoldModels;
```

Comment:
```suggestion
  const sortedModels = useMemo(() => {
     return [...models].sort((a, b) => sortModels(a, b, localeCode))
  }, [models, localeCode]);
  
  const visibleModels = useMemo(() => {
    return showAll
      ? sortedModel
      : sortedModels.slice(0, MAX_COLLAPSED_MODELS)
  }, [sortedModels, showAll]);
```

I think this would be preferable for two reason:
1. sort mutates the underlying array, so it's best to copy first
2. slice shallow copies, but then we reconstruct sortedModels by doing `[...aboveFoldModels, ...belowFoldModels]`. this way we can avoid the extra copies.